inference-engine/src/mkldnn_plugin/nodes/psroi.cpp

```diff
@@ -46,22 +50,36 @@ class PSROIPoolingImpl: public ExtLayerBase {
             part_size_ = layer->GetParamAsInt("part_size", 1);
             trans_std_ = layer->GetParamAsFloat("trans_std", 1);
 
+            Precision inputPrecision = layer->insData[0].lock()->getTensorDesc().getPrecision();
+
             if (no_trans_) {
-                addConfig(layer, {DataConfigurator(ConfLayout::PLN), DataConfigurator(ConfLayout::PLN)}, {DataConfigurator(ConfLayout::PLN)});
+                addConfig(layer, {DataConfigurator(ConfLayout::PLN, inputPrecision), DataConfigurator(ConfLayout::PLN, Precision::FP32)},
+                                 {DataConfigurator(ConfLayout::PLN, Precision::FP32)});
+
+                addConfig(layer, {DataConfigurator(ConfLayout::BLK16, inputPrecision), DataConfigurator(ConfLayout::PLN, Precision::FP32)},
+                                 {DataConfigurator(ConfLayout::PLN, Precision::FP32)});
             } else {
-                addConfig(layer, {DataConfigurator(ConfLayout::PLN), DataConfigurator(ConfLayout::PLN),
-                                  DataConfigurator(ConfLayout::PLN)}, {DataConfigurator(ConfLayout::PLN)});
+                addConfig(layer, {DataConfigurator(ConfLayout::PLN, inputPrecision), DataConfigurator(ConfLayout::PLN, Precision::FP32),
+                                  DataConfigurator(ConfLayout::PLN, Precision::FP32)}, {DataConfigurator(ConfLayout::PLN, Precision::FP32)});
+
+                addConfig(layer, {DataConfigurator(ConfLayout::BLK16, inputPrecision), DataConfigurator(ConfLayout::PLN, Precision::FP32),
+                                  DataConfigurator(ConfLayout::PLN, Precision::FP32)}, {DataConfigurator(ConfLayout::PLN, Precision::FP32)});
             }
         } catch (InferenceEngine::details::InferenceEngineException &ex) {
             errorMsg = ex.what();
         }
     }
 
-    StatusCode execute(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs,
-                       ResponseDesc *resp) noexcept override {
-        float* dst_data = outputs[0]->buffer();
-        const float *bottom_data_beginning = inputs[0]->buffer();
+    template <typename inputType, typename outputType>
+    StatusCode executeSpecified(std::vector<Blob::Ptr>& inputs, std::vector<Blob::Ptr>& outputs,
+                                ResponseDesc *resp) {
+        const auto *src_data = inputs[0]->cbuffer().as<const inputType*>();
         const float *bottom_rois_beginning = inputs[1]->buffer();
+        auto *dst_data = outputs[0]->buffer().as<outputType*>();
+
+        const int block_size = inputs[0]->getTensorDesc().getLayout() == Layout::BLOCKED ?
+                         inputs[0]->getTensorDesc().getBlockingDesc().getBlockDims()[4] : 1;
```

Comment:
Just to make sure that 4th index is always correct please add check in ctr that input/output should have rank 4.

Reply:
Will be done in the special PR. PSROI rollbacked.